### ISSUE TITLE
lomrf wlearn complains about predicate not defined

### ISSUE THREAD

**Issue by dportabella** [anskarl/LoMRF#6]
```
$ wget http://alchemy.cs.washington.edu/data/citeseer/citeseer.ie.tar.Z
$ tar -xvzf ../citeseer.ie.tar.Z
$ wget http://alchemy.cs.washington.edu/mlns/ie/ie_base.mln
$ query=InField/3,SameBib/2
$ lomrf wlearn -i ie_base.mln -t cs0-const.mln,cs0.db -o ie_base-learned.mln -ne $query
[ERROR] The predicate with signature 'FollowBy/3' is not defined.
```

However, FollowBy/3 is indeed defined. See:
```
$ head ie_base.mln
Token(token,position,bib)
FollowBy(bib,position,token)	// tkn[.|,] ...
Next(position,position)		// Next(j,i): j is next to i: j=i+1
...
```

What can be the problem?

**Comment by ewerlopes** [comment]
Hi.

I get a same error. Do you have any tips on solving it?

Thanks.

**Comment by dportabella** [comment]
not yet

**Comment by vagmcs** [comment]
Hello,

The problem appears to be the comment in line FollowBy(bib,position,token) // tkn[.|,] ... of the KB. If you delete the "..." from the comment the parser works fine. That means that the KB parser misses some case and does not ignore the comment properly. Nevertheless after you delete the "..." there is an assertion error stating that some ID in the internal evidence database is negative. The problem is that LoMRF indexes every ground atom in its internal evidence database in order to perform various optimizations. The indexes are currently integer numbers. But in this case the number of ground atoms are much more than the Int.MaxValue and the indexes are overflowing resulting in negative numbers. To solve this please split the data in the *.db file into smaller chuncks (multiple db files) and perform online weight learning using either AdaGrad or CDA. This should resolve the issue for now.

**Comment by ewerlopes** [comment]
Hi @dportabella 

Together with @luca-morreale we have discovered that the problem comes from the . on the comments. Additionally, the parser is not able to remove comments aside predicates. We have done a pull request that would solve the . problem, but having comments aside predicates is still a problem.

**Comment by anskarl** [comment]
I am closing this issue after merging the fix of @ewerlopes, pull request #9 'Fixed regex that parse comments'.
